cli/tests/test_diffedit_command.rs

```diff
@@ -110,6 +110,65 @@ fn test_diffedit() {
     "###);
 }
 
+#[test]
+fn test_diffedit_new_file() {
+    let mut test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    std::fs::write(repo_path.join("file1"), "a\n").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["new"]);
+    std::fs::remove_file(repo_path.join("file1")).unwrap();
+    std::fs::write(repo_path.join("file2"), "b\n").unwrap();
+
+    let edit_script = test_env.set_up_fake_diff_editor();
+
+    // Test the setup; nothing happens if we make no changes
+    std::fs::write(
+        &edit_script,
+        "files-before file1\0files-after JJ-INSTRUCTIONS file2",
+    )
+    .unwrap();
+    let stdout = test_env.jj_cmd_success(&repo_path, &["diffedit"]);
+    insta::assert_snapshot!(stdout, @r###"
+    Nothing changed.
+    "###);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["diff", "-s"]);
+    insta::assert_snapshot!(stdout, @r###"
+    R file1
+    A file2
+    "###);
+
+    // Creating `file1` on the right side is noticed by `jj diffedit`
+    std::fs::write(&edit_script, "write file1\nmodified\n").unwrap();
+    let stdout = test_env.jj_cmd_success(&repo_path, &["diffedit"]);
+    insta::assert_snapshot!(stdout, @r###"
+    Created rlvkpnrz 7b849299 (no description set)
+    Working copy now at: rlvkpnrz 7b849299 (no description set)
+    Parent commit      : qpvuntsm 414e1614 (no description set)
+    Added 1 files, modified 0 files, removed 0 files
+    "###);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["diff", "-s"]);
+    insta::assert_snapshot!(stdout, @r###"
+    M file1
+    A file2
+    "###);
+
+    // Creating a file that wasn't on either side is ignored by diffedit.
+    // TODO(ilyagr) This behavior is potentially a minor BUG.
+    test_env.jj_cmd_success(&repo_path, &["undo"]);
```

Comment:
Just a comment. It might not be intentional, but I think not including new files is also good since GUI diff tool can write backup files. I don't have strong opinion, though.

Reply:
Good point; I'll mention this in the comment. I don't have a strong opinion either, I mainly want to document it. I did waste almost an hour debugging this, thinking it was a bug in the 3-pane PR.